onnxruntime/core/optimizer/not_where_fusion.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/not_where_fusion.h"
+
+#include "core/graph/graph_utils.h"
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/utils.h"
+
+using namespace ONNX_NAMESPACE;
+using namespace onnxruntime::common;
+namespace onnxruntime {
+
+/**
+Transform that fuses two Not -> Where nodes to a single Where node
+with the where inputs 1 and 2 flipped.
+Condition ->  Not -> Where ->
+              value0-|  |
+              value1----|
+
+Condition -> Where ->
+      value1-|  |
+      value0----|
+ */
+Status NotWhereFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
+  GraphViewer graph_viewer(graph);
+  const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
+
+  for (auto node_index : node_topology_list) {
```

Comment:
Is it possible to iterate using graph.Nodes(), which doesn't require GraphViewer?

Reply:
changing to rewrite rule